providers/postgres/pyproject.toml

```diff
@@ -83,6 +83,9 @@ dependencies = [
 "psycopg" = [
     "psycopg[binary]>=3.2.9",
 ]
+"psycopg-c" = [
+    "psycopg[c]>=3.2.9",
+]
```

Comment:
To summarise my view from the discussion on Slack;

1. The binary package works fine in production, based on past experience I don't trust the psygopg dev's views on binary packages. I have been using the binary package of v3 in production at a scale larger than any individual airflow deployment fine for two years.

    So I don't see the need to change the prod image right now. Why change was isn't broken? A "but it performs better" claim needs number to back it.

1. I don't see that an extra is worth it. Just document "you can install the c extension if you want" since we have to document the presence of extras anyway.

Reply:
(I don't like the proliferation of extras, it causes regular headaches for us, and is difficult to remove in the future. We rarely do.)